atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/SweepCommand.java

```diff
@@ -0,0 +1,170 @@
+            final String table = entry.getKey();
+            Optional<byte[]> startRow = entry.getValue();
+
+            final AtomicLong cellsExamined = new AtomicLong();
+            final AtomicLong cellsDeleted = new AtomicLong();
+
+            while (startRow.isPresent()) {
+                Stopwatch watch = Stopwatch.createStarted();
+                SweepResults results = sweepRunner.run(table, config.getSweepBatchSize(), startRow.get());
+                System.out.println(String.format("Swept from %s to %s in table %s in %d ms, examined %d unique cells, deleted %d cells.",
+                        encodeStartRow(startRow), encodeEndRow(results.getNextStartRow()),
+                        table, watch.elapsed(TimeUnit.MILLISECONDS),
+                        results.getCellsExamined(), results.getCellsDeleted()));
+                startRow = results.getNextStartRow();
+                cellsDeleted.addAndGet(results.getCellsDeleted());
+                cellsExamined.addAndGet(results.getCellsExamined());
+                try {
+                    Thread.sleep(config.getSweepPauseMillis());
+                } catch (InterruptedException e) {
+                    throw Throwables.rewrapAndThrowUncheckedException(e);
+                }
+            }
+
+            services.getTransactionManager().runTaskWithRetry(new TxTask() {
+                @Override
+                public Void execute(Transaction t) {
+                    SweepPriorityTable priorityTable = SweepTableFactory.of().getSweepPriorityTable(t);
+                    SweepPriorityTable.SweepPriorityRow row = SweepPriorityTable.SweepPriorityRow.of(table);
+                    priorityTable.putWriteCount(row, 0L);
+                    priorityTable.putCellsDeleted(row, cellsDeleted.get());
+                    priorityTable.putCellsExamined(row, cellsExamined.get());
+                    priorityTable.putLastSweepTime(row, System.currentTimeMillis());
+
+                    System.out.println(String.format("Finished sweeping %s, examined %d unique cells, deleted %d cells.",
+                            table, cellsExamined.get(), cellsDeleted.get()));
+
+                    if (cellsDeleted.get() > 0) {
+                        Stopwatch watch = Stopwatch.createStarted();
+                        services.getKeyValueService().compactInternally(table);
+                        System.out.println(String.format("Finished performing compactInternally on %s in %d ms.",
+                                table, watch.elapsed(TimeUnit.MILLISECONDS)));
+                    }
+                    return null;
+                }
+            });
+        }
+        return 0;
+	}
+
+    private String encodeStartRow(Optional<byte[]> rowBytes) {
+        if (rowBytes.isPresent()) {
+            return BaseEncoding.base16().encode(Arrays.copyOf(rowBytes.get(), 12));
+        }
+        return BaseEncoding.base16().encode(Arrays.copyOf(new byte[0], 12));
```

Comment:
It is important that things are padded to 12 bytes in all cases (if someone provides more than 12 bytes this should probably error out, but if they provide fewer it should upgrade their array gracefully).  That said, there can still be simplification in the empty array case.